packages/purpose-process/test/testClonePurpose.ts

```diff
@@ -0,0 +1,304 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import {
+  decodeProtobufPayload,
+  getMockAgreement,
+  getMockEService,
+  getMockPurpose,
+  getMockPurposeVersion,
+  getMockTenant,
+  readLastEventByStreamId,
+  writeInReadmodel,
+} from "pagopa-interop-commons-test/index.js";
+import {
+  Purpose,
+  PurposeClonedV2,
+  agreementState,
+  generateId,
+  purposeVersionState,
+  tenantKind,
+  toPurposeV2,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { describe, expect, it, vi } from "vitest";
+import { formatDateAndTime } from "pagopa-interop-commons";
+import {
+  duplicatedPurposeTitle,
+  purposeCannotBeCloned,
+  purposeNotFound,
+  tenantKindNotFound,
+} from "../src/model/domain/errors.js";
+import { addOnePurpose } from "./utils.js";
+import {
+  agreements,
+  postgresDB,
+  purposeService,
+  purposes,
+  tenants,
+} from "./purposeService.integration.test.js";
+
+export const testClonePurpose = (): ReturnType<typeof describe> =>
+  describe("clonePurpose", async () => {
+    it("should write on event-store for the cloning of a purpose", async () => {
+      vi.useFakeTimers();
+      vi.setSystemTime(new Date());
+
+      const mockTenant = {
+        ...getMockTenant(),
+        kind: tenantKind.PA,
+      };
+      const mockEService = getMockEService();
+
+      const mockAgreement = getMockAgreement(
+        mockEService.id,
+        mockTenant.id,
+        agreementState.active
+      );
+
+      const mockPurpose: Purpose = {
+        ...getMockPurpose(),
+        eserviceId: mockEService.id,
+        consumerId: mockTenant.id,
+        versions: [getMockPurposeVersion(purposeVersionState.active)],
+      };
+
+      await addOnePurpose(mockPurpose, postgresDB, purposes);
+      await writeInReadmodel(mockTenant, tenants);
+      await writeInReadmodel(mockAgreement, agreements);
+
+      const { purpose } = await purposeService.clonePurpose({
+        purposeId: mockPurpose.id,
+        organizationId: mockTenant.id,
+        seed: {
+          eserviceId: mockEService.id,
+        },
+        correlationId: generateId(),
+      });
+
+      const writtenEvent = await readLastEventByStreamId(
+        purpose.id,
+        "purpose",
+        postgresDB
+      );
+
+      expect(writtenEvent).toMatchObject({
+        stream_id: purpose.id,
+        version: "0",
+        type: "PurposeCloned",
+        event_version: 2,
+      });
+
+      const writtenPayload = decodeProtobufPayload({
+        messageType: PurposeClonedV2,
+        payload: writtenEvent.data,
+      });
+
+      const expectedPurpose: Purpose = {
+        ...mockPurpose,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
In the tests we can safely use it on the whole file if we need it more than once

Reply:
Updated